### ISSUE TITLE
Console.log in concurrent browser print both on first test suite

### ISSUE THREAD

**Issue by dirkrooijakkers** [DevExpress/testcafe#3429]
### What is your Test Scenario?
Running two concurrent browsers with both 1 test case inside the suite. Both tests log a console message
 
### What is the Current behavior?
Both messages are print after Suite one is printed in the console.
 
### What is the Expected behavior?
Each message groups with the fixture
 
### What is your web application and your TestCafe test code?
```
fixture`Browser 1`
    .page('http://localhost:5000')
    .beforeEach( async t => {
        await t.resizeWindow(1024,690).setTestSpeed(0.8)})

    test('test1', async (t) => {
        console.log('   ↓ This is the leading browser');
    });

fixture`Browser 2`
    .page('http://localhost:5000')
    .beforeEach( async t => {
        await t.resizeWindow(1024,690).setTestSpeed(0.8)})

    test('test2', async (t) => {
        console.log('   ↓ This is the following browser');
    });
```
</details>
test
<details>
<summary>Your complete test report:</summary>
```
Browser 1

   ↓ This is the leading browser

   ↓ This is the following browser

 √ test1

 Browser 2
 √ test2
```

### Steps to Reproduce:
yarn testcafe -c 2 chrome testFolder
 
### Your Environment details:
 
* testcafe version:                   1.0.0
* node.js version:                    v10.14.2
* command-line arguments:  yarn testcafe -c 2 chrome testFolder
* browser name and version: Chrome 71.0.3578
* platform and version:          Windows 10.0.0

**Comment by dirkrooijakkers** [comment]
Edit: fixture`Browser 2`.beforeEach is not executing

**Comment by LavrovArtem** [comment]
This behavior is by design.
Tests are executed in parallel and test finishing is strictly determined by the order from the file. It helps to avoid changing reporters due to non-deterministic performance and test run speed in different browsers.
As for the `beforeEach` functionality, I suggest you create a separate issue if you have a problem with.
